### PR TITLE
[ResourceIsolation] analyze WorkGroup crud stmt

### LINKED ISSUE #2848: FE support CRUD of WorkGroup in FE
## Enhancement

WorkGroup is a concept that describe resource isolation in query evaluation, queries are classified into several perdefined workgroups according to its origination information, such as:
1. user, who issue this query?
2. role, which role the user belongs to?
3. query_type, what type is this query?
4. source_ip, from where this query is issued?

The queries' origination information is defined as workgroup classifier.

A User can manipulate a WorkGroup as follows.
1. create a workgroup.
2. drop a workgroup
3. alter a workgroup
4. show all workgroup
5. describe a workgroup

Notice: workgroup is internal term, for users perception, resource_group is used instead.

### LINKED ISSUE #3445: Analyze Workgroup stmts
## Feature request

Workgroup concept, to see https://github.com/StarRocks/starrocks/issues/2848
Workgroup stmt syntax parsing, to see https://github.com/StarRocks/starrocks/issues/3440

Analyze workgroup stmts before handle them.

for an example:
```
create resource_group rg1
to 
    (user='rg1_user1', role='rg1_role1', query_type in ('select', 'insert'), source_ip='192.168.2.1/24'),
    (user='rg1_user2', query_type in ('mv'), source_ip='192.168.3.1/24'),
    (user='rg1_user3', source_ip='192.168.4.1/24'),
    (user='rg1_user4')
with (
    'cpu_core_limit' = '10',
    'mem_limit' = '20%',
    'concurrency_limit' = '11',
    'type' = 'normal'
);
```
**classifier component**:
1. user: eq BinaryPredicate
2. role: eq BinaryPredicate
3. query _type: InPredicate
4. source_ip:  eq Binarypredicate

**resource quota**:
1. cpu_core_imit:  postive integer, never exceeds the average number of cpu cores per BE.
2. mem_limit: fraction between (0.0, 1.0), or percent between (0%, 100%).
3. concurrency_limit: positive integer.
4. type: normal|realtime

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

https://github.com/StarRocks/starrocks/issues/2848
https://github.com/StarRocks/starrocks/issues/3445

### CODE REVIEW

**Comment by kangkaisen** [comment]
please remove BE part from this PR

**Comment by satanson** [comment]
> please remove BE part from this PR

need rebase

**Review by kangkaisen — DISMISSED** [review]

**Review by laotan332 — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroup.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        WorkGroup workGroup = (WorkGroup) o;
+        return id == workGroup.id && version == workGroup.version;
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroup.java:185]: Why only compare id and version?

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupClassifier.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            classifiersStr.append(", query_type in (" + queryTypesStr + ")");
+        }
+        if (sourceIp != null) {
+            classifiersStr.append(", source_ip=" + sourceIp);
+        }
+        classifiersStr.append(")");
+        return classifiersStr.toString();
+    }
+
+    public enum QueryType {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupClassifier.java:165]: Why not StmtType?

File: fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.catalog;
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupAnalyzer.java:3]: WorkGroupAnalyzer.java should be in com.starrocks.sql.analyzer

File: fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupClassifier.java (hunk 2 above)
Comment (satanson) [reply fe/fe-core/src/main/java/com/starrocks/catalog/WorkGroupClassifier.java:165]: QueryTypes are defined by product manager.

**Review by Seaven — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**